parsers/EIA.py

```diff
@@ -435,7 +435,22 @@ def fetch_production_mix(zone_key, session=None, target_datetime=None, logger=No
             #replace small negative values (>-5) with 0s This is necessary for solar
             point = validate(point, logger=logger, remove_negative=True)
         mixes.append(mix)
-
+    
+    # Some of the returned mixes could be for older timeframes.
+    # Fx the latest oil data could be 6 months old.
+    # In this case we want to discard the old data as we won't be able to merge it
+    timeframes = [
+        [min(mix, key=lambda x: x['datetime'])['datetime'], max(mix, key=lambda x: x['datetime'])['datetime']]
+        for mix in mixes
+    ]
+    latest_timeframe = max(timeframes, key=lambda x: x[1])
+    
+    mixes = [
```

Comment:
```
correct_mixes = []
for mix in mixes:
  correct_mix = []
  for production_in_mix in mix:
    if production_in_mix['datetime'] >= latest_timeframe[0]:
      correct_mix.append(production_in_mix)
  if len(correct_mix) > 0:
    correct_mixes.append(correct_mix)
```

Reply:
it would actually be better to check that `production_in_mix['datetime'] in time_range` with `time_range` being the list of datetimes in the mix with latest datetimes